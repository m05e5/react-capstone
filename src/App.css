

ul {
  list-style-type: none;
  padding: 0;
}

.nav {
  display: flex;
  justify-content: space-between;
  padding: 0 1rem 0 1rem;
  background-color: rgb(211, 29, 114);
}

.navBack {
  display: flex;
  align-items: center;
  justify-content: center;
  font-weight: bold;
  text-decoration: none;
  color: white;
}

.navTitle {
  color: white;
  font-size: 16px;
}

.navAction {
  margin: 1rem 0 0 0;
  color: white;
}

.navIcon {
  padding: 0 5px 0 0;
}

.countries {
  display: grid;
  grid-template-columns: auto auto;
  margin: 0;
}

.country {
  background-color: rgb(211, 29, 114);
  position: relative;
  height: 200px;
  display: flex;
}

.dark {
  background-color: rgb(196, 0, 91);
}

.light {
  background-color: rgb(211, 29, 114);
}

.countryName {
  height: 100%;
  width: 100%;
  position: absolute;
  color: white;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
}

.countryName h2 {
  margin: 0;
}

.countryName p {
  margin: 0;
}

.countryImg {
  height:100%;
  width: 100%;
  opacity: 0.2;
  position: absolute;
}

.arrow {
  padding: 10px;
}

.regions {
  margin: 0;
  color: white;
}

.region {
  display: flex;
  padding: 5px;
  align-items: center;
  justify-content: space-between;
}

