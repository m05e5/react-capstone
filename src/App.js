import './App.css';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from './components/Home';
import store from './redux/configureStore';
import Regions from './components/Regions';

function App() {
  return (
    <div className="App">
      <header className="App-header" />
      <Router>
        <Switch>
          <Route exact path="/">
            <Provider store={store}>
              <Home />
            </Provider>
          </Route>
          <Route path="/regions">
            <Provider store={store}>
              <Regions />
            </Provider>
          </Route>
        </Switch>
      </Router>

    </div>
  );
}

export default App;
