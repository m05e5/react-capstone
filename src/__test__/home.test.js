import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

const ReactTestRenderer = require('react-test-renderer');

it('Creates a snapshot of the home page and it should match the previous one', () => {
  const tree = ReactTestRenderer.create(<Provider store={store}><App /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
