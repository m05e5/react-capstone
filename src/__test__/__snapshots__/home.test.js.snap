// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a snapshot of the home page and it should match the previous one 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  />
  <div>
    <div
      className="nav"
    >
      <a
        aria-current="page"
        className="navBack active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="navIcon"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 16"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.224 1.553a.5.5 0 0 1 .223.67L6.56 8l2.888 5.776a.5.5 0 1 1-.894.448l-3-6a.5.5 0 0 1 0-.448l3-6a.5.5 0 0 1 .67-.223z"
            fillRule="evenodd"
          />
        </svg>
        <p>
          2022
        </p>
      </a>
      <p
        className="navTitle"
      >
        Hi-Coro
      </p>
      <div
        className="navAction"
      >
        <svg
          className="navIcon"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 16"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
          />
          <path
            d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
          />
        </svg>
        <svg
          className="navIcon"
          fill="currentColor"
          height={20}
          viewBox="0 0 16 16"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
          />
        </svg>
      </div>
    </div>
    <div
      className="intro"
    >
      <img
        alt="corona"
        className="introImg"
        src="corona.jpg"
      />
      <div
        className="introDetails"
      >
        <h2>
          Covid 19
        </h2>
        <p>
          Number of cases
        </p>
        <p>
          in 2020-03-22
        </p>
      </div>
    </div>
    <p
      className="mid"
    >
      STATES BY COUNTRY
    </p>
    <ul
      className="countries"
    />
  </div>
</div>
`;
